frontend/src/components/file-explorer/FileExplorer.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import {
+  IoIosArrowBack,
+  IoIosArrowForward,
+  IoIosRefresh,
+} from "react-icons/io";
+import { twMerge } from "tailwind-merge";
+import { getWorkspace } from "../../services/fileService";
+import ExplorerTree from "./ExplorerTree";
+import { removeEmptyNodes } from "./utils";
+import IconButton from "../IconButton";
+
+interface ExplorerActionsProps {
+  onRefresh: () => void;
+  toggleHidden: () => void;
+  isHidden: boolean;
+}
+
+function ExplorerActions({
+  toggleHidden,
+  onRefresh,
+  isHidden,
+}: ExplorerActionsProps) {
+  return (
+    <div
+      className={twMerge(
+        "transform flex h-[24px] items-center gap-1 absolute top-4 right-2",
+        isHidden ? "right-3" : "right-2",
+      )}
+    >
+      {!isHidden && (
+        <IconButton
+          icon={
+            <IoIosRefresh
+              size={20}
+              className="text-neutral-400 hover:text-neutral-100 transition"
+            />
+          }
+          testId="refresh"
+          ariaLabel="Refresh workspace"
+          onClick={onRefresh}
+        />
+      )}
+
+      <IconButton
+        icon={
+          isHidden ? (
+            <IoIosArrowForward
+              size={20}
+              className="text-neutral-400 hover:text-neutral-100 transition"
+            />
+          ) : (
+            <IoIosArrowBack
+              size={20}
+              className="text-neutral-400 hover:text-neutral-100 transition"
+            />
+          )
+        }
+        testId="close"
+        ariaLabel="Close workspace"
+        onClick={toggleHidden}
+      />
+    </div>
+  );
+}
+
+interface FileExplorerProps {
+  onFileClick: (path: string) => void;
+}
+
+function FileExplorer({ onFileClick }: FileExplorerProps) {
+  const [workspace, setWorkspace] = React.useState<TreeNode[]>([]);
+  const [isHidden, setIsHidden] = React.useState(false);
+
+  const getWorkspaceData = async () => {
+    const wsFile = await getWorkspace();
+    setWorkspace(removeEmptyNodes([wsFile]));
```

Comment:
Maybe we should not transmit empty notes in the beginning. Then we could get rid of this.

Reply:
It would definitely improve things given the *current state*. It is like this because there was only one way of differentiating between a folder and a file: if it has children. Current setup checks for exactly the same thing but checks for `node.children` compared to the previous `node.children.length`.

Of course, neither approach holds when considering the existence of *empty files*